### PR TITLE
fix metric alias

### PR DESCRIPTION
**Pull request by guolinke**
refer to https://github.com/microsoft/LightGBM/pull/2209#issuecomment-513459716

### CODE REVIEW

**Comment by StrikerRUS** [comment]
@guolinke Is it possible to share `std::string GetMetricType(const std::string& type)` function for objective too? It seems to be that the difference is in `l2`/`l2_root` and `multiclass`/`multiclassova` only.

**Comment by StrikerRUS** [comment]
@guolinke I updated docs accordingly.

**Comment by guolinke** [comment]
Sorry, didn't get it. Could explain with more details?

**Comment by StrikerRUS** [comment]
@guolinke I mean, we have practically the same code for objective function here
https://github.com/microsoft/LightGBM/blob/master/src/objective/objective_function.cpp as for metric function which you've updated in this PR. So I wonder can we use `std::string GetMetricType(const std::string& type)` for replacing aliases with main names for both metrics and objectives.

**Comment by StrikerRUS** [comment]
@guolinke In other words, replace 
https://github.com/microsoft/LightGBM/blob/7189743d224a405bb48050834460d107fb7c1c0b/src/io/config.cpp#L70
with
```
Metric::ParseMetrics(value, objective);
```

This will allow to work only with objectives names in further code, without considering aliases.

**Comment by guolinke** [comment]
@StrikerRUS Thanks, good point! will update it accordingly.

**Comment by StrikerRUS** [comment]
@guolinke Don't you want to merge `GetMetricType` and `ParseObjective` functions into one function (e.g. `ParseFunctionAliases`)? And that function can take one additional enum argument `function_type` with two possible values `metric` and `objective` to help with `l2`/`l2_root` and `multiclass`/`multiclassova` cases, which are different in two current functions.

**Comment by guolinke** [comment]
@StrikerRUS I tried it, and think it is more easy-to-read to use two functions.
Actually, the duplicated aliases are no many, just two or three.
 but introducing the function_type, with caring about the duplicate or not, will increase the complexity of this function.

**Comment by StrikerRUS** [comment]
@guolinke OK, got it!

File: include/LightGBM/objective_function.h [hunk 1]
```diff
@@ -88,6 +88,7 @@ class ObjectiveFunction {
   * \brief Load objective function from string object
   */
   LIGHTGBM_EXPORT static ObjectiveFunction* CreateObjectiveFunction(const std::string& str);
+
```
Comment (StrikerRUS) [inline include/LightGBM/objective_function.h]: Excess diff.

File: src/io/config.cpp [hunk 2]
```diff
@@ -3,6 +3,8 @@
  * Licensed under the MIT License. See LICENSE file in the project root for license information.
  */
 #include <LightGBM/config.h>
+#include <LightGBM/metric.h>
+#include <LightGBM/objective_function.h>
```
Comment (StrikerRUS) [inline src/io/config.cpp]: I guess that these includes are not needed anymore.

File: src/io/config.cpp [hunk 3]
```diff
@@ -63,41 +65,88 @@ void GetBoostingType(const std::unordered_map<std::string, std::string>& params,
   }
 }
 
+std::string ParseObjectiveAlias(const std::string& type) {
+  if (type == std::string("regression") || type == std::string("regression_l2")
+    || type == std::string("mean_squared_error") || type == std::string("mse")
+    || type == std::string("l2_root") || type == std::string("root_mean_squared_error") || type == std::string("rmse")) {
+    return "regression";
```
Comment (StrikerRUS) [inline src/io/config.cpp]: Maybe add `l2` alias here (and in docs)?

File: src/io/config.cpp [hunk 4]
```diff
@@ -63,41 +65,88 @@ void GetBoostingType(const std::unordered_map<std::string, std::string>& params,
...
+  } else if (type == std::string("regression_l1") || type == std::string("mean_absolute_error") || type == std::string("mae")) {
+    return "regression_l1";
```
Comment (StrikerRUS) [inline src/io/config.cpp]: Maybe add `l1` alias here (and in docs)?

File: src/io/config.cpp [hunk 5]
```diff
@@ -63,41 +65,88 @@ void GetBoostingType(const std::unordered_map<std::string, std::string>& params,
...
+    return "cross_entropy_lambda";
+  } else if (type == std::string("mean_absolute_percentage_error") || type == std::string("mape")) {
+    return "mape";
+  } else if (type == std::string("none") || type == std::string("null") || type == std::string("custom") || type == std::string("na")) {
+    return "custom";
+  }
+  return type;
+}
+
+std::string ParseMetricAlias(const std::string& type) {
```
Comment (StrikerRUS) [inline src/io/config.cpp]: Add aliases for `custom` too?

**Comment by StrikerRUS** [comment]
I updated docs according to new objective names.

**Comment by StrikerRUS** [comment]
I think that the following code can be simplified by removing checks for aliases, since we have already done it earlier:

https://github.com/microsoft/LightGBM/blob/7189743d224a405bb48050834460d107fb7c1c0b/src/io/config.cpp#L198-L234

**Comment by StrikerRUS** [comment]
I added new aliases for regression objectives in docs.

**Comment by StrikerRUS** [comment]
@guolinke Can we merge?

**Comment by guolinke** [comment]
Yes

**Pull request merged**